get FunctionalTensorWrapper to propagate symints

bdhirsh opened the pull request:
`FunctionalTensorWrapper` needs to know when it's dealing with ints vs. SymInts.

I'm pretty sure the right way to do this is:
```
if (has_symbolic_sizes_strides())
   set_sym_sizes_and_strides(..);
} else {
   set_sizes_and_strides(..);
}
```

Stack from [ghstack](https://github.com/ezyang/ghstack):
* #81302
* __->__ #81301
* #81238

Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

bdhirsh committed:
get FunctionalTensorWrapper to propagate symints

[ghstack-poisoned]

bdhirsh committed:
Update on "get FunctionalTensorWrapper to propagate symints"




[ghstack-poisoned]

bdhirsh committed:
Update on "get FunctionalTensorWrapper to propagate symints"




[ghstack-poisoned]

bdhirsh committed:
Update on "get FunctionalTensorWrapper to propagate symints"




[ghstack-poisoned]

bdhirsh committed:
Update on "get FunctionalTensorWrapper to propagate symints"




[ghstack-poisoned]

ezyang commented on aten/src/ATen/FunctionalTensorWrapper.cpp:
```diff
@@ -183,9 +185,20 @@ void FunctionalTensorWrapper::replace_(const Tensor& other) {
+      refresh_contiguous();
+    }
+  } else {
+    // For sym ints, this will check pointer equality, which I'm pretty sure is what we want.
```
yes. Importantly, the pointer equality test will never falsely say that the sizes are the same when they are not

ezyang reviewed (approved):
would be nice to get symbolic numel going. Contiguity is going to be trouble too.

bdhirsh committed:
Update on "get FunctionalTensorWrapper to propagate symints"

`FunctionalTensorWrapper` needs to know when it's dealing with ints vs. SymInts.

I'm pretty sure the right way to do this is:
```
if (has_symbolic_sizes_strides())
   set_sym_sizes_and_strides(..);
} else {
   set_sizes_and_strides(..);
}
```




[ghstack-poisoned]

albanD reviewed (commented):
Can we easily do basic testing for this?

bdhirsh committed:
Update on "get FunctionalTensorWrapper to propagate symints"

`FunctionalTensorWrapper` needs to know when it's dealing with ints vs. SymInts.

I'm pretty sure the right way to do this is:
```
if (has_symbolic_sizes_strides())
   set_sym_sizes_and_strides(..);
} else {
   set_sizes_and_strides(..);
}
```




[ghstack-poisoned]

bdhirsh committed:
Update on "get FunctionalTensorWrapper to propagate symints"

`FunctionalTensorWrapper` needs to know when it's dealing with ints vs. SymInts.

I'm pretty sure the right way to do this is:
```
if (has_symbolic_sizes_strides())
   set_sym_sizes_and_strides(..);
} else {
   set_sizes_and_strides(..);
}
```




[ghstack-poisoned]

bdhirsh committed:
Update on "get FunctionalTensorWrapper to propagate symints"

`FunctionalTensorWrapper` needs to know when it's dealing with ints vs. SymInts.

I'm pretty sure the right way to do this is:
```
if (has_symbolic_sizes_strides())
   set_sym_sizes_and_strides(..);
} else {
   set_sizes_and_strides(..);
}
```




[ghstack-poisoned]

bdhirsh committed:
Update on "get FunctionalTensorWrapper to propagate symints"

`FunctionalTensorWrapper` needs to know when it's dealing with ints vs. SymInts.

I'm pretty sure the right way to do this is:
```
if (has_symbolic_sizes_strides())
   set_sym_sizes_and_strides(..);
} else {
   set_sizes_and_strides(..);
}
```




[ghstack-poisoned]

bdhirsh committed:
Update on "get FunctionalTensorWrapper to propagate symints"

`FunctionalTensorWrapper` needs to know when it's dealing with ints vs. SymInts.

I'm pretty sure the right way to do this is:
```
if (has_symbolic_sizes_strides())
   set_sym_sizes_and_strides(..);
} else {
   set_sizes_and_strides(..);
}
```




[ghstack-poisoned]

bdhirsh committed:
Update on "get FunctionalTensorWrapper to propagate symints"

`FunctionalTensorWrapper` needs to know when it's dealing with ints vs. SymInts.

I'm pretty sure the right way to do this is:
```
if (has_symbolic_sizes_strides())
   set_sym_sizes_and_strides(..);
} else {
   set_sizes_and_strides(..);
}
```




[ghstack-poisoned]

bdhirsh committed:
Update on "get FunctionalTensorWrapper to propagate symints"

`FunctionalTensorWrapper` needs to know when it's dealing with ints vs. SymInts.

I'm pretty sure the right way to do this is:
```
if (has_symbolic_sizes_strides())
   set_sym_sizes_and_strides(..);
} else {
   set_sizes_and_strides(..);
}
```




[ghstack-poisoned]

bdhirsh committed:
Update on "get FunctionalTensorWrapper to propagate symints"

`FunctionalTensorWrapper` needs to know when it's dealing with ints vs. SymInts.

I'm pretty sure the right way to do this is:
```
if (has_symbolic_sizes_strides())
   set_sym_sizes_and_strides(..);
} else {
   set_sizes_and_strides(..);
}
```




[ghstack-poisoned]

bdhirsh committed:
Update on "get FunctionalTensorWrapper to propagate symints"

`FunctionalTensorWrapper` needs to know when it's dealing with ints vs. SymInts.

I'm pretty sure the right way to do this is:
```
if (has_symbolic_sizes_strides())
   set_sym_sizes_and_strides(..);
} else {
   set_sizes_and_strides(..);
}
```




[ghstack-poisoned]

bdhirsh committed:
Update on "get FunctionalTensorWrapper to propagate symints"

`FunctionalTensorWrapper` needs to know when it's dealing with ints vs. SymInts.

I'm pretty sure the right way to do this is:
```
if (has_symbolic_sizes_strides())
   set_sym_sizes_and_strides(..);
} else {
   set_sizes_and_strides(..);
}
```




[ghstack-poisoned]

zhxchen17 commented:
@zhxchen17 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/).

bdhirsh committed:
Update on "get FunctionalTensorWrapper to propagate symints"

`FunctionalTensorWrapper` needs to know when it's dealing with ints vs. SymInts.

I'm pretty sure the right way to do this is:
```
if (has_symbolic_sizes_strides())
   set_sym_sizes_and_strides(..);
} else {
   set_sizes_and_strides(..);
}
```


Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

[ghstack-poisoned]

bdhirsh committed:
Update on "get FunctionalTensorWrapper to propagate symints"

`FunctionalTensorWrapper` needs to know when it's dealing with ints vs. SymInts.

I'm pretty sure the right way to do this is:
```
if (has_symbolic_sizes_strides())
   set_sym_sizes_and_strides(..);
} else {
   set_sizes_and_strides(..);
}
```


Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

[ghstack-poisoned]

bdhirsh committed:
Update on "get FunctionalTensorWrapper to propagate symints"

`FunctionalTensorWrapper` needs to know when it's dealing with ints vs. SymInts.

I'm pretty sure the right way to do this is:
```
if (has_symbolic_sizes_strides())
   set_sym_sizes_and_strides(..);
} else {
   set_sizes_and_strides(..);
}
```


Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

[ghstack-poisoned]

bdhirsh committed:
Update on "get FunctionalTensorWrapper to propagate symints"

`FunctionalTensorWrapper` needs to know when it's dealing with ints vs. SymInts.

I'm pretty sure the right way to do this is:
```
if (has_symbolic_sizes_strides())
   set_sym_sizes_and_strides(..);
} else {
   set_sizes_and_strides(..);
}
```


Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

[ghstack-poisoned]

bdhirsh committed:
Update on "get FunctionalTensorWrapper to propagate symints"

`FunctionalTensorWrapper` needs to know when it's dealing with ints vs. SymInts.

I'm pretty sure the right way to do this is:
```
if (has_symbolic_sizes_strides())
   set_sym_sizes_and_strides(..);
} else {
   set_sizes_and_strides(..);
}
```


Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

[ghstack-poisoned]
